pool/example_test.go

```diff
@@ -122,49 +142,15 @@ func ExampleConnectionPool_NewWatcher() {
 	time.Sleep(time.Second)
 }
 
-func ExampleConnectionPool_NewWatcher_noWatchersFeature() {
-	const key = "foo"
```

Comment:
The example could fail if servers does not support the feature.

Reply:
As it its connection pool, we just ignore such connections.

Reply:
Could we test it?

Reply:
Yes, test was added.
https://github.com/tarantool/go-tarantool/blob/d185407a7997d6901ed7ec41b980dd3b3c9bbe68/pool/connection_pool_test.go#L2876-L2899